pages/docs.tsx

```diff
@@ -0,0 +1,27 @@
+import { GetStaticProps, InferGetStaticPropsType } from 'next';
+import { createSwaggerSpec } from 'next-swagger-doc';
+import SwaggerUI from 'swagger-ui-react';
+import 'swagger-ui-react/swagger-ui.css';
+
+const ApiDoc = ({ spec }: InferGetStaticPropsType<typeof getStaticProps>) => {
+  return <SwaggerUI spec={spec} />;
+};
+
+export const getStaticProps: GetStaticProps = async (ctx) => {
+  const spec: Record<string, any> = createSwaggerSpec({
+    definition: {
+      openapi: '3.0.0',
+      info: {
+        title: 'NextJS Swagger',
+        version: '0.2.0',
+      },
+    },
+  });
```

Comment:
`any` should be avoided when possible. Here, it clearly doesn't matter, but its a bad habit in typescript.

The real power of gradual typing is that *if* all modules are well typed then the whole program is well typed.

`any` is usually a precursor to some gross type casting, since you can go from `any` to any other type. Keeping our domain types "sound" then we genuinely trust that the static types prohibit type errors at runtime. If not, then typescript is nothing more than a fancy docstring saying what a type _should_ be, but not what it actually _is_.

For more, ask me for my matthias/typescript speech 😄

Reply:
thanks Ryan! makes sense. we're keeping the any in the annotation *reference the convo in Slack* but thank you!